lms/djangoapps/shoppingcart/models.py

```diff
@@ -1168,6 +1169,16 @@ class CourseRegistrationCode(models.Model):
     invoice = models.ForeignKey(Invoice, null=True)
     invoice_item = models.ForeignKey(CourseRegistrationCodeInvoiceItem, null=True)
 
+    @classmethod
+    def get_registration_code(cls, code, course_id):
```

Comment:
The lookup by `course_id` is not necessary since the `code` field is unique across all `CourseRegistrationCode` objects. That argument should be dropped. Looking at the views, you are really only making use of the `None` return value twice. Given the limited usage, I would move the exception handling to those views and get rid of this method.